File: content/snippets/cpp/language_basics/exceptions_and_recovery.cpp
#include <iostream>
#include <string>

int parse_port(const std::string& value) {
    return std::stoi(value);
}

int main() {
    try {
        int port = parse_port("not-a-number");
        std::cout << "port: " << port << "\n";
    } catch (const std::exception& ex) {
        std::cout << "parse failed: " << ex.what() << "\n";
        std::cout << "fallback port: 8080\n";
    }

    std::cout << "program continues\n";
    return 0;
}

File: content/snippets/go/language_basics/exceptions_and_recovery.go
package main

import (
	"fmt"
	"strconv"
)

func parsePort(value string) (int, error) {
	return strconv.Atoi(value)
}

func main() {
	port, err := parsePort("not-a-number")
	if err != nil {
		fmt.Println("parse failed:", err)
		fmt.Println("fallback port:", 8080)
	} else {
		fmt.Println("port:", port)
	}

	fmt.Println("program continues")
}
